File: client/src/entities/Task/model/types/task.ts
import { ProgrammingLanguage } from 'shared/consts/ProgrammingLanguage.ts';

export interface TaskSolutionInfo {
    id: string;
    taskOrder: string;
    taskName: string;
    taskText: string;
    startCode: string;
    isDone: boolean;
    runAttemptsLeft: number;
    programmingLanguage: ProgrammingLanguage;
}

export interface TaskInfo {
    id: string;
    name: string;
    taskText: string;
    programmingLanguage: ProgrammingLanguage;
    startCode: string;
    testsCode: string;
    runAttempts: number;
    isDeleted?: boolean;
}

File: CodeRev/UserService.DAL/Entities/Task.cs
using UserService.DAL.Models.Enums;

namespace UserService.DAL.Entities
{
    public class Task : BaseEntity
    {
        public bool IsDeleted { get; set; }
        public string TaskText { get; set; }
        public string StartCode { get; set; }
        public string Name { get; set; }
        public string TestsCode { get; set; }
        public int RunAttempts { get; set; }
        public ProgrammingLanguage ProgrammingLanguage { get; set; }
    }
}